src/Bundle/spec/Sender/Adapter/SymfonyMailerAdapterSpec.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\MailerBundle\Sender\Adapter;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Mailer\Event\EmailSendEvent;
+use Sylius\Component\Mailer\Model\EmailInterface;
+use Sylius\Component\Mailer\Renderer\RenderedEmail;
+use Sylius\Component\Mailer\Sender\Adapter\AbstractAdapter;
+use Sylius\Component\Mailer\SyliusMailerEvents;
+use Symfony\Component\Mailer\Exception\TransportException;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+final class SymfonyMailerAdapterSpec extends ObjectBehavior
+{
+    function let(MailerInterface $mailer): void
+    {
+        $this->beConstructedWith($mailer);
+    }
+
+    function it_is_an_adapter(): void
+    {
+        $this->shouldHaveType(AbstractAdapter::class);
+    }
+
+    function it_sends_an_email_with_events(
+        MailerInterface $mailer,
+        EmailInterface $email,
+        EventDispatcherInterface $dispatcher,
+        RenderedEmail $renderedEmail
+    ): void {
+        $this->setEventDispatcher($dispatcher);
+
+        $renderedEmail->getSubject()->willReturn('subject');
+        $renderedEmail->getBody()->willReturn('body');
+
+        $dispatcher
+            ->dispatch(Argument::type(EmailSendEvent::class), SyliusMailerEvents::EMAIL_PRE_SEND)
+            ->shouldBeCalled()
+        ;
+
+        $mailer->send(Argument::type(Email::class))->shouldBeCalled();
```

Comment:
We could work a little bit at this prophecy to check the Email content as well 🖖